docs/guides/software-lifecycle/security/security-scanning/README.md

```diff
@@ -0,0 +1,238 @@
+
+   ```bash
+   vi ...
+   ```
+
+Additional steps such as customizing reports and/or enabling further automation are covered in detail below.
+
+---
+
+## Step-by-Step Guide
+
+SCRUB may be run locally or as a CI workflow action, such as in GitHub Actions. Please see below sections for further details. 
+
+### Table of Contents
+- [Security Scanning](#security-scanning)
+  - [Introduction](#introduction)
+  - [Prerequisites](#prerequisites)
+  - [Quick Start](#quick-start)
+  - [Step-by-Step Guide](#step-by-step-guide)
+    - [Table of Contents](#table-of-contents)
+    - [Client-side Scan and Audit](#client-side-scan-and-audit)
+      - [Steps](#steps)
+    - [GitHub.com Actions Analysis on Push and Pull Request](#githubcom-actions-analysis-on-push-and-pull-request)
+      - [Steps](#steps-2)
+    - [Frequently Asked Questions (FAQ)](#frequently-asked-questions-faq)
+  - [Credits](#credits)
+  - [Feedback and Contributions](#feedback-and-contributions)
+
+### Client-side Scan and Audit
+The developer's local environment is scanned directly using the `SCRUB` tool. After scanning, a report containing detected security issues is generated. Developers can audit this report for detailed information on detected security concerns.
+
+#### Steps
+1. **Installation**
+   - Install the release version of  [SCRUB](https://nasa.github.io/scrub/installation.html).
+     ```bash
+     pip3 install --upgrade --user nasa-scrub
+     ```
+
+2. **Scanning**
+   - Scan all local files from the current directory and output results to a baseline file.
+     ```bash
+     detect-secrets scan --all-files --disable-plugin AbsolutePathDetectorExperimental --exclude-files '\.secrets.*' --exclude-files '\.git*' > .secrets.baseline
+     ```
+
+3. **Checking Results**
+   - View the results in the baseline file.
+     ```bash
+     ...
+     ```
+
+4. **Analysis**
+   - Analyze results using the `audit` tool.
+     ```bash
+     ...
+     ```
+
+[View more on advanced SCRUB scan configuration](https://nasa.github.io/scrub/configuration-inputs.html)
+
+> ℹ️ **Note**: Any confirmed security issues should be addressed and mitigated before pushing to remote repositories.
+
+### GitHub.com Actions Analysis on Push and Pull Request
+
+Code is scanned for security risks within the repository. It leverages [GitHub Action](https://github.com/features/actions). The scan is triggered during a push or pull request and any detected security vulnerabilities are reported while blocking merges or pushes to protected branches.
+
+#### Steps
+1. **Workflow Creation**
+   - The first step is to create a `codeql.yaml` workflow file in the `.github/workflows` directory to define the GitHub action. Copy and paste the below while ensuring the correct branch of your codebase is referenced. For example (from the [Slim Python Starter Kit](https://github.com/NASA-AMMOS/slim-starterkit-python/blob/main/.github/workflows/codeql.yml)):
+  
+```yaml
```

Comment:
Indentation here would be good.

Reply:
Indentation ... Do you mean with the code sample or text?

Reply:
@riverma Indentation ... Do you mean with the code sample or text?

Reply:
@ingyhere - sorry - meant the code (yaml) sample block. It is easier to read if it's indented with the text prior to it.